crates/resolver/src/name_server/name_server.rs

```diff
@@ -48,6 +52,7 @@ impl<P: ConnectionProvider> NameServer<P> {
         config: NameServerConfig,
         options: Arc<ResolverOpts>,
         tls: Arc<TlsConfig>,
+        encrypted_transport_state: Arc<AsyncMutex<NameServerTransportState>>,
```

Comment:
Hmm, I don't think we want to expose `futures_util::lock::Mutex` in public API. Also, should really see if we can wrap up `ResolverOpts`, `tls` and this thing together.

Reply:
I think you could stick it in the new `PoolContext`.

- https://github.com/hickory-dns/hickory-dns/pull/3297

Reply:
> I don't think we want to expose futures_util::lock::Mutex in public API

Fixed.

> I think you could stick it in the new PoolContext.

Done.